packages/wallet/src/SingleAddressWallet/SingleAddressWallet.ts

```diff
@@ -131,15 +133,23 @@ export const DEFAULT_POLLING_CONFIG = {
   pollInterval: 5000
 };
 
+const isCardanoTx = (input: Cardano.Tx | Cardano.SerializedTx): input is Cardano.Tx => {
+  const { id, body, witness } = input as Cardano.Tx;
+  return id !== undefined && body !== undefined && witness !== undefined;
+};
+
+const isSerializedTx = (input: Cardano.Tx | Cardano.SerializedTx): input is Cardano.SerializedTx =>
+  typeof (input as Cardano.SerializedTx) === 'string';
```

Comment:
This is a bit confusing. We already have `isCardanoTx` narrowing predicate, which, if negated, says that the input is `Cardano.SerializedTx`. However, the `isSerializedTx` narrowing predicate takes a different verification approach.
Ideally, it should be like this:
```diff
-  typeof (input as Cardano.SerializedTx) === 'string';
+  !isCardanoTx(input);

Reply:
It now has 3 variants and 2 narrowing functions: https://github.com/input-output-hk/cardano-js-sdk/pull/600/commits/b31b69edec251b41788a866bcc3e4dd06530fa48#diff-aeb17003218a7a98485f6c851ac5be47b6876308204497ddd74a552f69fa9393R130-R156

Please review again